pkg/helm/helm.go

```diff
@@ -87,14 +87,11 @@ func (h *Hydrator) templateArgs(ctx context.Context, destDir string) ([]string,
 }
 
 func (h *Hydrator) appendValuesArgs(args []string) ([]string, error) {
-	var values map[string]interface{}
-	err := json.Unmarshal([]byte(h.Values), &values)
-	if err != nil {
-		return []string{}, fmt.Errorf("failed to unmarshal helm.values, error: %w", err)
-	}
-	for key, val := range values {
-		args = append(args, "--set", fmt.Sprintf("%s=%v", key, val))
+	valuesPath := "tmp/chart-values.yaml"
+	if err := ioutil.WriteFile(valuesPath, []byte(h.Values), 0644); err != nil {
+		return []string{}, fmt.Errorf("failed to create values file: %w", err)
 	}
+	args = append(args, "--values", valuesPath)
```

Comment:
Do we need to clean-up/remove the temp file after we consume it later on?

Reply:
Per other comments, no clean up necessary it seems.

Reply:
Current solution is creating and overwriting a specific value file. We will not create more than one temp file under the temp folder. So I think it is ok to not keep removing the temp file. And since it is a temp file, it will be removed when it is not needed.